src/tools/rbd_mirror/PoolReplayer.cc

```diff
@@ -1127,7 +1145,7 @@ void PoolReplayer<I>::handle_remote_pool_meta_updated(
     const RemotePoolMeta& remote_pool_meta) {
   dout(5) << "remote_pool_meta=" << remote_pool_meta << dendl;
 
-  if (!m_default_namespace_replayer) {
+  if (!m_init_done) {
```

Comment:
Are you sure this is an equivalent transformation?  The existing code bails from `handle_remote_pool_meta_updated()` if replayer for the default namespace isn't running.  The assumption appears to be that since there is nothing running there is nothing to stop as failure to start the replayer for the default namespace means that `m_pool_replayer_thread` isn't even created.  The new code flips this boolean to true at some point in `PoolReplayer::init()` but it can't know anything about whether the replayer for the default namespace is running.